src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp

```diff
@@ -106,7 +106,8 @@ class boss_baron_rivendare : public CreatureScript
         void AttackStart(Unit* who) override
         {
             //can't use entercombat(), boss' dmg aura sets near players in combat, before entering the room's door
-            instance->SetData(TYPE_BARON, IN_PROGRESS);
+            if (instance && instance->GetData(TYPE_BARON) == NOT_STARTED)
```

Comment:
There is no need to check 'instance' for nullptr, it will never be empty - the AI will simply not run if spawned outside the instance, thanks to GetInstanceAI<boss_baron_rivendareAI>